typescript/lib/services/service_context.ts

```diff
@@ -65,42 +66,48 @@ class ServiceContext {
     }
 
     // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
-    public execCommand(command: string[]): [number, Uint8Array | string, Error] {
+    public async execCommand(command: string[]): Promise<[number, Uint8Array | string, Error]> {
         const serviceId: ServiceID = this.serviceId;
         const args: ExecCommandArgs = newGetExecCommandArgs(serviceId, command);
 
-        var resp: ExecCommandResponse;
-        var error: grpc.ServiceError;
-        this.client.execCommand(args, function(error, feature){ //TODO (comment) - don't use return value, but use callback parameters
-            error = error; 
-            resp = feature;
+        const promiseAsync: Promise<ResultAsync<ExecCommandResponse, Error>> = new Promise((resolve, _unusedReject) => {
+            this.client.execCommand(args, (_unusedError: grpc.ServiceError, response: ExecCommandResponse) => {
+                resolve(okAsync(response));
+            })
         });
-        
-        //TODO (comment) - Will this error checking propogate as needed (Kevin mentioned that returning error as-is is good enough)
-        if (error !== null) {
-            return [0, null, error]; 
-            //TODO - passing in actual error, but no personalized message
-            //I was thinking of `new Error("An error occurred executing command " + command +  " on service " + serviceId + 
-            //". Here is the error message: " + error)` but don't think this propogates the error as needed, just instantiates an error
+
+        const promise: Result<ExecCommandResponse, Error> = await promiseAsync;
```

Comment:
this actually isn't a promise - it's just the result (I'd probably call this something like `execCmdResult` or whatnot)